xrpl/core/binarycodec/main.py

```diff
@@ -73,6 +71,28 @@ def encode_for_signing_claim(json: Dict[str, Any]) -> str:
     return buffer.hex().upper()
 
 
+def encode_for_signing_batch(json: Dict[str, Any]) -> str:
+    """
+    Encode a Batch transaction's data to be signed.
+
+    Args:
+        json: A JSON-like dictionary representation of Batch data.
+
+    Returns:
+        The binary-encoded Batch data, ready to be signed.
+    """
+    prefix = _BATCH_PREFIX
+    flags = UInt32.from_value(json["flags"])
+    transaction_ids = json["transaction_ids"]
+    len_transaction_ids = UInt32.from_value(len(transaction_ids))
+
+    buffer = prefix + bytes(flags) + bytes(len_transaction_ids)
+    for tx in transaction_ids:
+        buffer += bytes(Hash256.from_value(tx))
+
+    return buffer.hex().upper()
```

Comment:
Where can I find the description of the serialization/deserialization algorithm for the `Batch` transaction?

I didn't observe it in the [spec](https://github.com/XRPLF/XRPL-Standards/discussions/162)

More generally, is this an accepted norm for encoding new transaction-types? I'm asking about points like how to determine order of the fields, what data types can we use for each of the fields, etc.

For instance, the length of the transaction-IDs could have been represented with a UINT8 (since its at most 8 elements) and a UINT32. How did you decide on one type?

Reply:
https://github.com/XRPLF/XRPL-Standards/tree/master/XLS-0056d-batch#2132-signingpubkey-and-txnsignature